tests/test_circleci_auto_approve.py

```
@@ -0,0 +1,173 @@
+        "type": "approval",
+        "status": "blocked",
+        "approval_request_id": APPROVAL_ID,
+      }
+    ]
+  }
+
+
+def test_allowlisted_user_approves_without_membership_lookup() -> None:
+  calls: list[tuple[str, str]] = []
+
+  def urlopen(request: Any, timeout: int = 20) -> FakeResponse:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `request`

(ANN401)